app/javascript/createObserver.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Initializes a DOM mutation observer that will run a hook when
+ * a DOM node matching a selector is mounted or unmounted.
+ *
+ *   const myObs = createObserver({
+ *     selector: "[data-ace-editor]",
+ *     onMount: node => {},
+ *     onUnmount: node => {}
+ *   });
+ */
+
+const DEFAULT_ROOT_ELEMENT = document;
+const DEFAULT_OBSERVER_CONFIG = { childList: true, subtree: true };
+
+export const createObserver = config => {
+  const {
+    observerConfig = DEFAULT_OBSERVER_CONFIG,
+    rootElement = DEFAULT_ROOT_ELEMENT,
+    selector,
+    onMount,
+    onUnmount
+  } = config;
+
+  const observer = new MutationObserver(mutations => {
+    mutations.forEach(mutation => {
+      // Handle added nodes
+      if (onMount) {
+        mutation.addedNodes.forEach(addedNode => {
+          const matchingElements = getMatchingElementsFromTree(
+            addedNode,
+            selector
+          );
+          if (matchingElements.length < 1) return;
+          matchingElements.forEach(node => onMount(node));
+        });
+      }
+      // Handle removed nodes
+      if (onUnmount) {
```

Comment:
Maybe we don't need to copy the entire file? eg looks like we are not using this if.

Reply:
We are not but it could be useful, especially if we use more of that